src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -694,19 +687,26 @@ public static ImmutableConciseSet doIntersection(Iterator<ImmutableConciseSet> s
     return new ImmutableConciseSet(IntBuffer.wrap(retVal.toArray()));
   }
 
-  private static void sort(final WordIterator[] a, final int to, final Comparator<WordIterator> comp)
+  /**
+   * Variation of insertion sort, elements [maxChangedIndex + 1, size) are sorted, elements [0, maxChangedIndex] should
+   * be inserted into that sorted range.
+   */
+  private static void partialSort(
```

Comment:
`partialSort()` is now used only for intersection. The maximum number of bitmaps to intersect that happened during last 2 weeks in our production is 31. Theoretical limit is the number of dimensions in data sources.